CMakeLists.txt

```diff
@@ -5,10 +5,13 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 option(BUILD_SHARED_LIBS "Build shared libraries" OFF)
 
 include(FetchContent)
-FetchContent_Declare(SFML
+FetchContent_Declare(sfml
     GIT_REPOSITORY https://github.com/SFML/SFML.git
     GIT_TAG 2.6.x)
-FetchContent_MakeAvailable(SFML)
+if (NOT sfml_POPULATED)
+    FetchContent_Populate(sfml)
```

Comment:
This has actually been deprecated: https://cmake.org/cmake/help/latest/policy/CMP0169.html

With [CMake 3.28](https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_declare:~:text=of%20the%20effects.-,New%20in%20version%203.28%3A,-EXCLUDE_FROM_ALL) there's a much simpler solution now:

```cmake
include(FetchContent)
FetchContent_Declare(SFML
    GIT_REPOSITORY https://github.com/SFML/SFML.git
    GIT_TAG 2.6.x
    EXCLUDE_FROM_ALL)
FetchContent_MakeAvailable(SFML)
```

Reply:
I"m a big fan of `EXCLUDE_FROM_ALL` in my `FetchContent` calls. It's not unreasonable to raise this template's CMake requirement to 3.28. All the latest versions of macOS, Windows, and Ubuntu include this version or CMake or newer, but it will cause errors for Ubuntu 22 users, however many of such users there are.

Reply:
Yeah mentioned that in the original description - if we're happy updating though I'll do that